### ISSUE TITLE
🗺️ Navigation Blocking

### ISSUE THREAD

**Issue by ryanflorence** [remix-run/react-router#9714]

**Comment by ryanflorence** [comment]
@ryanflorence to mess w/ the demo

**Comment by brophdawg11** [comment]
I pushed up a little blocking demo app in `examples/navigation-blocking` in the `history-blocking` branch from https://github.com/remix-run/react-router/pull/9709.  If you check that branch out: 

```bash
> yarn clean
> yarn
> cd examples/navigation-blocking
> npm ci
> USE_SOURCE=true npm run dev
```

It has an index page, then pages `one`/`two`/`three`/`four`/`five` pages and a form using a blocker on `three`.  Best way to test is to start a fresh tab, click through the pages up to `five` sequentially, then back button to page `three`.

Then you have a history stack of `[index, one, two, three*, four, five]` and can start playing with blocking from there.

**Comment by chaance** [comment]
Shipped in `6.7.0-pre.3`

**Comment by brophdawg11** [comment]
Now that the prerelease is out, you can just check out `release-next` and run the following for the above demo app:

```
cd examples/navigation-blocking
npm ci
npm run dev
```
